### ISSUE TITLE
Snapshot support for Event Store

### ISSUE THREAD

**Issue by nowakprojects** [johanhaleby/occurrent#73]
It's may be nice to allow composition of event store, which will decorate current EventStore which behaviour which support snapshots of stream state.

**Comment by johanhaleby** [comment]
Yeah, that would be nice. But when you say snapshot here, do you mean a "snapshot event" and not a "snapshot document/blob"? I think the first makes the most sense and it would allow for composability with the same "EventStore" interface I think (what I think you refer to [here](https://github.com/johanhaleby/occurrent/issues/73)).

We should schedule another meeting someday to flush out some ideas :)

**Comment by nowakprojects** [comment]
Yes, I thought about SnapshotEvent. This approach will probably fit into current flow :) 

I'm working with the code and do not want to forget some ideas, so I try to adding issues without delay. Those only just enhancments propositions :)

**Comment by johanhaleby** [comment]
Available in 0.31.0
